server/src/main/java/org/elasticsearch/index/seqno/SeqNoStats.java

```diff
@@ -83,6 +83,24 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
         return builder;
     }
 
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        final SeqNoStats that = (SeqNoStats) o;
+        return maxSeqNo == that.maxSeqNo &&
+            localCheckpoint == that.localCheckpoint &&
+            globalCheckpoint == that.globalCheckpoint;
+    }
+
+    @Override
+    public int hashCode() {
+        int result = Long.hashCode(maxSeqNo);
+        result = 31 * result + Long.hashCode(localCheckpoint);
+        result = 31 * result + Long.hashCode(globalCheckpoint);
+        return result;
```

Comment:
Since this is only going to be used in tests, I think we can get away with:

```suggestion
        return Objects.hash(maxSeqNo, localCheckpoint, globalCheckpoint);
```

Reply:
Yep, I pushed https://github.com/elastic/elasticsearch/pull/35223/commits/869d265d4aa68fb248b1cfc996f6676e889e27ba.